src/riscv/lib/src/machine_state/memory.rs

```diff
@@ -49,6 +50,23 @@ pub const PAGE_SIZE: NonZeroU64 = {
     }
 };
 
+#[expect(
+    clippy::reversed_empty_ranges,
+    reason = "An empty range is expected with zero length"
+)]
+/// Calculate the page range for a given address and length.
+pub fn get_page_range(address: Address, length: usize) -> std::ops::RangeInclusive<usize> {
+    let address = address as usize;
+    let start_page = address >> OFFSET_BITS;
+
+    if length == 0 {
```

Comment:
Why don't you accept `length: NonZeroUsize` like we discussed in the weekly meeting?